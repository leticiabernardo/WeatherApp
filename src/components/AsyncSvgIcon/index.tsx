import { useDynamicSVGImport } from '@/hooks/asyncSVG';

type AsyncSvgIconProps = {
  svg: string;
  options?: {
    fill: string;
    width: string;
    height: string;
  };
};

const defaultProps = {
  options: {
    fill: 'white',
    width: '35px',
    height: '35px',
  },
};

const AsyncSvgIcon = ({ svg, options }: AsyncSvgIconProps) => {
  const { SvgIcon } = useDynamicSVGImport(svg, {});

  return SvgIcon ? (
    <SvgIcon
      fill={options?.fill}
      width={options?.width}
      height={options?.height}
    />
  ) : (
    <div />
  );
};
AsyncSvgIcon.defaultProps = defaultProps;

export default AsyncSvgIcon;
