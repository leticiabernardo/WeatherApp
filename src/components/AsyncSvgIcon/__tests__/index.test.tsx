import { render, cleanup, waitFor } from '@testing-library/react';
import AsyncSvgIcon from '../index';

describe('default wrapper render', () => {
  afterEach(cleanup);

  test('render the component', async () => {
    const { container } = await waitFor(() =>
      render(<AsyncSvgIcon svg="teste" />)
    );
    expect(container).toMatchSnapshot();
  });
});
