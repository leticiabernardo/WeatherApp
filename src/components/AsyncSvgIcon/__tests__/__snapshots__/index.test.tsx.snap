// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default wrapper render render the component 1`] = `
<div>
  <div
    class="svg-mock"
  />
</div>
`;
